blockchain/blockdao/blockdao.go

```diff
@@ -54,23 +54,45 @@ type (
 	}
 
 	blockDAO struct {
-		blockStore   BlockDAO
-		indexers     []BlockIndexer
-		timerFactory *prometheustimer.TimerFactory
-		lifecycle    lifecycle.Lifecycle
-		headerCache  cache.LRUCache
-		footerCache  cache.LRUCache
-		receiptCache cache.LRUCache
-		blockCache   cache.LRUCache
-		tipHeight    uint64
+		blockStore          BlockDAO
+		indexers            []BlockIndexer
+		timerFactory        *prometheustimer.TimerFactory
+		lifecycle           lifecycle.Lifecycle
+		headerCache         cache.LRUCache
+		footerCache         cache.LRUCache
+		receiptCache        cache.LRUCache
+		blockCache          cache.LRUCache
+		tipHeight           uint64
+		disableIndexerCheck bool
 	}
+
+	Option func(*blockDAO) error
 )
 
+func DisableCheckIndexerOption() Option {
+	return func(dao *blockDAO) error {
+		dao.disableIndexerCheck = true
```

Comment:
currently, the archive shard need those indexer files in gateway mode, like `bloomfilter.index.db, stake.index.db` to start the sever process.
this option should only disable check for those files, but keep check for sync indexers `trie.db, contractstaking.index.db`
by doing this, we don't need to manually copy those files to start the archive shard